engine/table/src/main/java/io/deephaven/engine/table/impl/util/FunctionGeneratedTableFactory.java

```diff
@@ -155,6 +163,15 @@ private FunctionBackedTable getTable() {
 
     private long updateTable() {
         Table newTable = tableGenerator.get();
+        if (newTable.isRefreshing()) {
+            if (ExecutionContext.getContext().getUpdateGraph() != newTable.getUpdateGraph()) {
+                throw new IllegalStateException(
+                        "Function-generated tables must belong to the same UpdateGraph as the creating FunctionGeneratedTableFactory.");
+            }
+            if (!newTable.getUpdateGraph().satisfied(newTable.getUpdateGraph().clock().currentStep())) {
+                throw new IllegalStateException("The function-generated table must be satisfied to be valid.");
```

Comment:
`"The function-generated table is not satisfied; did you miss a dependency when specifying source tables?"`